### PR TITLE
Scaling API changes

### LINKED ISSUE #7394: [Scaling API]: Meta-issue
Need to ensure the following: 

- [x] New ACLs are implemented and enforced on the new endpoints #7411
- [x] Scheduler respects new Min/Max fields #7412
- [x] Scaling events #7422
- [x] API fields are all present
- [x] API fields are all populated (i.e., scaling events)
- [x] blocking queries supported by all endpoints that should support it
- [x] More tests (some POC content may be missing tests)
- [x] CLI commands are implemented
- [x] New APIs are documented
- [x] New `scaling` block is documented

Lower priority issues:
- [x] stale queries supported by all endpoints that should support it #7410
- [x] No extraneous API fields are present

### PR DESCRIPTION
**Pull request by cgbaker**
partially completes #7394 

Includes:
- new `Scaling` block on `TaskGroup`
- Enforce policy min/max during job registration #7413 
- new RPCs: 
  - `Job.Scale`: short-cut for a Job., will be expanded later to also persist a scaling event
  - `Job.ScaleStatus`: get the current counts for a job's task group. similar to Job.Summary, but with more autoscaler-specific information and an appropriate ACL capability
  - `Scaling.ListPolicies`: list the autoscaling polices in a namespace
  - `Scaling.GetPolicy`: get a specific autoscaling policy
- ACL support and testing for the new RPC endpoints #7418 
- HTTP endpoints and API support for the above RPC endpoints
- new state store table, a view of the scaling policies contained in jobs
- lifecycle to populate the scaling policy table during job lifecycle

Currently missing (will address after PR): 
- CLI support
- documentation in website

### CODE REVIEW

File: client/taskenv/env.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ func (t *TaskEnv) All() map[string]string {
 }
 
 // AllValues is a map of the task's environment variables and the node's
-// attributes with cty.Value (String) values. Errors including keys are
+// attributes with cty.Count (String) values. Errors including keys are
```
Comment (cgbaker) [inline client/taskenv/env.go]:
```suggestion
// attributes with cty.Value (String) values. Errors including keys are
```

File: client/taskenv/util.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func addNestedKey(dst map[string]interface{}, k, v string) error {
 	return nil
 }
 
-// ctyify converts nested map[string]interfaces to a map[string]cty.Value. An
+// ctyify converts nested map[string]interfaces to a map[string]cty.Count. An
```
Comment (cgbaker) [inline client/taskenv/util.go]:
```suggestion
// ctyify converts nested map[string]interfaces to a map[string]cty.Value. An
```

File: client/taskenv/util_test.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ func TestAddNestedKey_Ok(t *testing.T) {
 		// M will be initialized if unset
 		M map[string]interface{}
 		K string
-		// Value is always "x"
+		// Count is always "x"
```
Comment (cgbaker) [inline client/taskenv/util_test.go]:
```suggestion
		// Value is always "x"
```

File: client/taskenv/util_test.go [hunk 4]
```diff
@@ -226,7 +226,7 @@ func TestAddNestedKey_Bad(t *testing.T) {
 		// M will be initialized if unset
 		M func() map[string]interface{}
 		K string
-		// Value is always "x"
+		// Count is always "x"
```
Comment (cgbaker) [inline client/taskenv/util_test.go]:
```suggestion
		// Value is always "x"
```

File: nomad/structs/structs.go [hunk 5]
```diff
@@ -4003,6 +4081,8 @@ func (j *Job) SpecChanged(new *Job) bool {
 	c.JobModifyIndex = j.JobModifyIndex
 	c.SubmitTime = j.SubmitTime
 
+	// cgbaker: FINISH: probably need some consideration of scaling policy ID here
+
```
Comment (cgbaker) [inline nomad/structs/structs.go]:
```suggestion
```

**Comment by cgbaker** [comment]
two tests failures are both flaky tests, merging now. will put Changelog in another PR

File: nomad/mock/mock.go [hunk 6]
```diff
@@ -1253,3 +1253,34 @@ func ACLManagementToken() *structs.ACLToken {
...
+		ID: uuid.Generate(),
+		Target: map[string]string{
+			structs.ScalingTargetNamespace: structs.DefaultNamespace,
+			structs.ScalingTargetJob:       uuid.Generate(),
+			structs.ScalingTargetGroup:     uuid.Generate(),
+		},
+		Policy: map[string]interface{}{
+			"a": "b",
+		},
+		Enabled:     true,
```
Comment (lgfa29) [inline nomad/mock/mock.go]: Add `Min` and `Max`

**Pull request merged**